sdk/digitaltwins/Azure.DigitalTwins.Core/tests/QueryTests.cs

```diff
@@ -97,13 +100,32 @@ public async Task Query_PaginationWorks()
                 // act
                 var options = new QueryOptions();
                 options.MaxItemsPerPage = pageSize;
-                AsyncPageable<string> asyncPageableResponse = client.QueryAsync(queryString, options);
+
+                CancellationTokenSource queryTimeoutCancellationToken = new CancellationTokenSource(QueryWaitTimeout);
+                bool queryHasExpectedCount = false;
+                while (!queryHasExpectedCount)
+                {
+                    if (queryTimeoutCancellationToken.IsCancellationRequested)
+                    {
+                        throw new AssertionException($"Timed out waiting for at least {pageSize + 1} twins to be queryable");
+                    }
+
+                    AsyncPageable<string> asyncPageableResponse = client.QueryAsync(queryString);
```

Comment:
do you want to pass the cancellation token into this or the AsPages methods?

Reply:
Sure, just in case